lib/client/profile.go

```diff
@@ -133,7 +133,11 @@ func (h *profiles) convertKVPairToAPI(d *model.KVPair) (unversioned.Resource, er
 	ap.Metadata.Labels = bp.Labels
 	ap.Spec.IngressRules = rulesBackendToAPI(bp.Rules.InboundRules)
 	ap.Spec.EgressRules = rulesBackendToAPI(bp.Rules.OutboundRules)
-	ap.Spec.Tags = bp.Tags
+	if len(bp.Tags) == 0 {
```

Comment:
In the meantime, could you move the code that is in combat.go into the convertAPIToKVPair() method above - that way the forward and reverse conversions are in the same place.